### ISSUE TITLE
[GOG] Cloud Save: "ValueError: month must be in 1..12" during manual saves download

### ISSUE THREAD

**Issue by d-rez** [Heroic-Games-Launcher/HeroicGamesLauncher#2541]
### Describe the bug

When I try to download (or force download after the initial download / game reinstall) my cloud saves for Witcher 3, it always errors out with the following error message

```
ValueError: month must be in 1..12
```

Syncing my Cyberpunk2077 save files works completely fine

### Add logs

```text
[SAVES] INFO: Local files: 65
[SAVES] INFO: Files in cloud: 141
[SAVES] WARNING: Forcing download
[SAVES] INFO: Downloading files
Traceback (most recent call last):
  File "gogdl/cli.py", line 50, in <module>
  File "gogdl/cli.py", line 46, in main
  File "gogdl/saves.py", line 160, in sync
  File "gogdl/saves.py", line 295, in download_file
ValueError: month must be in 1..12
[308] Failed to execute script 'cli' due to unhandled exception!
```

### Steps to reproduce

1. Go to Witcher 3 settings (GOG Version, GOTY Complete Edition)
2. Select "Force Download" and click on SYNC in cloud saves section
3. Wait

### Expected behavior

Saves download successfully without error

### Screenshots

![cloudsyncerror-witcher3](https://user-images.githubusercontent.com/1719659/225333114-8922a747-b4bf-4b36-9749-d2056430a4a1.png)

### Heroic Version

Latest Stable

### System Information

- Steam Deck
- Tried using Proton, Wine etc
- Cleared Heroic cache

### Additional information

Happy to troubleshoot further but Heroic logs were very scarce. Not sure how to debug this.

**Comment by imLinguin** [comment]
That's really odd. The part that's failing is not able to parse the last modified header. It's probably malformed in some way

**Comment by imLinguin** [comment]
Could you try the build from the mentioned PR?  
It should show us more values when the error happens.  
Also please backup your saves on another PC/Galaxy client, to avoid loosing them while playing arround with gogdl

1. Get the Linux binary
2. Extract it and mark it as executable
3. Move it to location accessible by Heroic (such as `~/Games/Heroic`)
4. Set the executable in Heroic advanced settings

Then try to reproduce the issue

**Comment by imLinguin** [comment]
Oh I didn't took that in account. It's the build compatible with not yet relased version of Heroic. You can try latest build 
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/actions/runs/4435221613 (flatpak)  
or  
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/actions/runs/4435221616 (AppImage)

**Comment by d-rez** [comment]
Thanks. Sorry for the delay, haven't had a whole lot of time for tinkering recently :p

Here's the updated error message:

```
[SAVES] INFO: Local files: 68
[SAVES] INFO: Files in cloud: 141
[SAVES] WARNING: Forcing download
[SAVES] INFO: Downloading files
Traceback (most recent call last):
  File "gogdl/cli.py", line 51, in <module>
  File "gogdl/cli.py", line 47, in main
  File "gogdl/saves.py", line 161, in sync
  File "gogdl/saves.py", line 295, in download_file
ValueError: month must be in 1..12
[267] Failed to execute script 'cli' due to unhandled exception!
Header 2022-00-03T00:11:23+00:00
File /home/deck/Games/Heroic/Prefixes/The Witcher 3 Wild Hunt - Complete Edition/dosdevices/c:/users/deck/Documents/The Witcher 3/gamesaves/ManualSave_10083c_7e600c00_2cb2c00.png
```

**Comment by imLinguin** [comment]
Well, there's nothing I can really do since the malformed header is sent by the server. I'll just make it to not crash on it I guess

**Comment by d-rez** [comment]
Yeah it does seem to be a server-side data error thing, I could probably try finding the save file from within the game and deleting it but if it's possible to have the sync not crash that's probably better (I have no idea what caused this header to be saved like this, but this means there's a chance that more people can encounter it)

Again thanks for looking into it :)

**Comment by arielj** [comment]
I understand this should be fixed in the new 2.7 release, right?

**Comment by imLinguin** [comment]
Yes

**Comment by d-rez** [comment]
I can confirm this is now fixed in 2.7 with the bundled gogdl v0.7.1

Now I'm just getting a bunch (10x to be exact) warnings about an incorrect LastModified header but cloud save download succeeds.

Thanks!
